#  <p style="text-align: center;">Практическое задание к Семинару 5.<br><br>Урок 5. Тестирование гипотез.</p>

<div class="alert alert-block alert-danger"><font size=4 color=black>
<b>Здравствуйте! Не успеваю, обязуюсь выложить все задания на GitHub до 12:00 (МСК) 29.10.2023 на GitHub. Заранее спасибо.</b></font></div>

In [11]:
import scipy.stats as stats

**АЛГОРИТМ МЕТОДИКИ ТЕСТИРОВАНИЯ  ГИПОТЕЗ**
<div class="alert alert-block alert-warning"><font size=4 color=black><b>1 этап - Формулирование нулевой  $H_{0}$ и  альтернативной гипотез $H_{1}$;<br>
2 этап - Выбор уровня статистической значимости α;<br>
3 этап - Выбор статистического критерия;<br>
4 этап - Расчет наблюдаемого критерия;<br>
5 этап - Сравнение табличного и наблюдаемого значений критерия;<br>
6 этап - Вывод.</b></font></div>

**Z-критерий**
<div class="alert alert-block alert-warning"><font size=4 color=black><b>$z_н=\frac{\bar X-\mu_0}{\sigma/\sqrt n}$ (1)</b></font></div>


Таблица 1. Z Таблица (положительные значения)

![Positive Z Table](https://datascience.eu/wp-content/uploads/2020/05/image1.png)

Таблица 2. Z Таблица (отрицательные значения)

![Negative Z Table](https://www.ztable.net/wp-content/uploads/2018/11/negativeztable.png)

## ЗАДАНИЕ 1. 
**Когда используется критерий Стьюдента, а когда Z –критерий?**

**<p style="text-align: center;">
    <font size=10 color=#32CD32><b>-------------------------------</b></font><br>
    <font size=4.5 color=#3ac444>РЕШЕНИЕ:</font><br>
<font size=10 color=#32CD32><b>-------------------------------</b></font></p>**

<div class="alert alert-block alert-success"><font size=3 color='b'><b>ОТВЕТ: </b><br>Критерии Стьюдента и Z-критерий, являются параметрическими критериями и используются для статистической проверки гипотезы путем сравнения двух выборочных средних. Оба критерия предполагают, что выборка взята из нормально распределенной генеральной совокупности, и имеется равенство дисперсий в группах. Если известна сигма генеральной совокупности более предпочтительнее ипспользовать Z- критерий, в противном случае -  критерий Стьюдента t. </font></div>

## ЗАДАНИЕ 2. 
**Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв. мм.**

**<p style="text-align: center;">
    <font size=10 color=#32CD32><b>-------------------------------</b></font><br>
    <font size=4.5 color=#3ac444>РЕШЕНИЕ:</font><br>
<font size=10 color=#32CD32><b>-------------------------------</b></font></p>**

**1 этап - Формулирование нулевой  $H_{0}$ и  альтернативной гипотез $H_{1}$.**<br>
***$H_{0}$ гипотеза*** - Средний диаметр шариков для подшипников, изготовленные автоматическим станком <b>НЕ ИМЕЕТ</b> различий с диаметром шиариков из выборки. $H_{0}$: $M(X) = 17 => \mu=\mu_0$<br>
***$H_{1}$ гипотеза*** - Средний диаметр шариков для подшипников, изготовленные автоматическим станком <b>ИМЕЕТ</b> различие с диаметром шиариков из выборки. $H_{1}$: $M(X) = 17,5 => \mu>\mu_0$<br><br>
**2 этап - Выбор уровня статистической значимости α.**<br>
Из условия **α = 0,05**.<br><br>
**3 этап - Выбор статистического критерия.**<br>
Т.к. из условия нам известна дисперсия, то будем использовать Z-критерий.<br><br>
**4 этап - Расчет наблюдаемого критерия.**<br>
***Дано:***<br>
$\alpha = 0.05$<br>
$D(X) = 4$<br>
$\sigma=\sqrt 4=2$<br>
n = 100 размер выборки<br>
$\bar X = \mu=17.5$<br>
$\mu_{0}=17$<br>
По формуле (1), находим наблюдаемое значение Z-критерия.

In [12]:
z=(17.5-17)/0.2
print(f'Наблюдаемое значение z-критерия: {z: .4f}')

Наблюдаемое значение z-критерия:  2.5000


**5 этап - Сравнение табличного и наблюдаемого значений критерия.**<br>
Табличное значение z-критерия. Т.к. α = 5% (0.05) и $\mu>\mu_0$, то альтернативная гипотеза находиться справа, а слева лежат 0.95 или 95% всех значений стандартного нормального распределения. По таблице Z-значений сначала ищем долю 0.95. А потом смотрим, на пересечении каких составных частей Z лежит эта доля. $Z_{tab}\approx1.645$.<br><br>

In [14]:
# Проверка правилности определения табличного значения
z_tab=stats.norm.ppf(0.95)
print(f'Табличное значение z-критерия: Z_tab {z_tab: .4f}')

Табличное значение z-критерия: Z_tab  1.6449


$Z_{nab} > Z_{tab} = > 2.5000 > 1.645 $ 

Т.к. $Z_{nab} > Z_{tab}$ отвергаем нулевую гипотезу на уровне значимости  𝛼=0.05 и можем констатировать, что средний диаметр шариков больше 17.

<div class="alert alert-block alert-success"><font size=3 color='b'><b>ОТВЕТ: </b><br> $Z_{nab} > Z_{tab}$ отвергаем нулевую гипотезу на уровне значимости  𝛼=0.05 и можем констатировать, что средний диаметр шариков больше 17.  </font></div>